ios/RNTPTDocumentView.m

```diff
@@ -876,6 +876,92 @@ - (void)saveDocumentWithCompletionHandler:(void (^)(NSString * _Nullable filePat
     }
 }
 
+#pragma mark - Annotation Flag
+
+- (void)setFlagForAnnotations:(NSArray *)annotationFlagList
+{
+    if (annotationFlagList.count == 0) {
+        return;
+    }
+    
+    PTPDFViewCtrl *pdfViewCtrl = self.pdfViewCtrl;
+    BOOL shouldUnlock = NO;
+    @try {
+        [pdfViewCtrl DocLock:YES];
+        shouldUnlock = YES;
```

Comment:
This way of locking is correct, but it's the "old way" of doing it, and we're not writing new code like this anymore. We have a couple of convenience methods that make this easier to write here (one that acquires a read lock and one that acquires a write lock): https://www.pdftron.com/api/ios/Classes/PTPDFViewCtrl.html#/Locking

You can see an example of its usage in the `deleteAnnotations:` method in this file. Could you please change any new code you've written to use the block-based locking? Thank you!